--- origsrc/libical-0.41/src/libicalss/icalspanlist_cxx.h	2008-10-29 10:20:08.000000000 -0500
+++ src/libical-0.41/src/libicalss/icalspanlist_cxx.h	2008-11-11 23:55:13.078754900 -0600
@@ -15,9 +15,12 @@
 #ifndef ICALSPANLIST_CXX_H
 #define ICALSPANLIST_CXX_H
 
+extern "C" {
 #include <libical/ical.h>
 #include <libicalss/icalset.h>
 #include <libicalss/icalspanlist.h>
+}
+
 #include "vcomponent.h"
 #include <vector>		/* For as_matrix.. */
 
