--- origsrc/libical-0.44/src/libicalss/Makefile.am
+++ src/libical-0.44/src/libicalss/Makefile.am
@@ -64,7 +64,7 @@
 	$(BDB_INCLUDE)
 
 libicalss_la_LDFLAGS = -version-info 44:0:44
-libicalss_la_LIBADD = $(BDB_LIBFILES) 
+libicalss_la_LIBADD = $(BDB_LIBFILES) ../libical/libical.la
 
 if DEV
 libicalss_la_DEVSOURCES = icalsslexer.l icalssyacc.y
@@ -114,6 +114,7 @@
 	$(srcdir)/icalmessage.h
 
 if WITH_CXX_BINDINGS
+libicalss_cxx_la_LIBADD = libicalss.la ../libical/libical_cxx.la
 libicalss_cxx_la_SOURCES =	\
 	$(libicalss_evolution_la_SOURCES)	\
 	icalspanlist_cxx.h	\
