--- origsrc/libical-0.31/configure.in	2008-03-12 20:50:18.000000000 -0500
+++ src/libical-0.31/configure.in	2008-05-04 03:15:40.265625000 -0500
@@ -92,8 +92,8 @@
   )
 
   AC_MSG_CHECKING(for Berkeley DB include files)
-  if test -f $BDB_DIR/include/db.h; then
-   BDB_DIR_INCLUDE=$BDB_DIR/include
+  if test -f $BDB_DIR/include/db4.0/db.h; then
+   BDB_DIR_INCLUDE=$BDB_DIR/include/db4.0
    AC_MSG_RESULT($BDB_DIR_INCLUDE)
   else
     if test -f $BDB_DIR/include/db4/db.h; then
@@ -306,8 +306,6 @@
 		  AC_SUBST(PY_LIB_LOC)
 		  AC_SUBST(PY_CFLAGS)
 		  AC_SUBST(PY_EXTRA_LIBS)
-		  # SWIG needs PIC in order to work.
-		  CFLAGS="$CFLAGS -fPIC"
 	  else
 		  python_val=false
 	  fi
--- origsrc/libical-0.31/src/libical/Makefile.am	2008-03-12 20:50:02.000000000 -0500
+++ src/libical-0.31/src/libical/Makefile.am	2008-05-04 02:38:54.015625000 -0500
@@ -52,7 +52,7 @@
 	-I$(top_builddir)/src/libical	\
 	-I$(srcdir)
 
-libical_la_LDFLAGS = -version-info 30:0:30
+libical_la_LDFLAGS = -version-info 30:0:30 -no-undefined
 
 libical_la_LIBADD = $(PTHREAD_LIBS)
 
@@ -109,6 +109,7 @@
 	astime.h 
 
 if WITH_CXX_BINDINGS
+libical_cxx_la_LDFLAGS = -no-undefined
 libical_cxx_la_SOURCES =	\
 	$(libical_la_SOURCES)	\
 	vcomponent.h		\
--- origsrc/libical-0.31/src/libicalss/Makefile.am	2008-03-12 20:50:05.000000000 -0500
+++ src/libical-0.31/src/libicalss/Makefile.am	2008-05-04 03:47:25.218750000 -0500
@@ -63,8 +63,8 @@
 	-I$(top_builddir)/src/libical	\
 	$(BDB_INCLUDE)
 
-libicalss_la_LDFLAGS = -version-info 30:0:30
-libicalss_la_LIBADD = $(BDB_LIBFILES) ../libical/.libs/libical.la
+libicalss_la_LDFLAGS = -version-info 30:0:30 -no-undefined
+libicalss_la_LIBADD = $(BDB_LIBFILES) ../libical/libical.la
 
 if DEV
 libicalss_la_DEVSOURCES = icalsslexer.l icalssyacc.y
@@ -114,8 +114,10 @@
 	$(srcdir)/icalmessage.h
 
 if WITH_CXX_BINDINGS
+libicalss_cxx_la_LDFLAGS = -no-undefined
+libicalss_cxx_la_LIBADD = $(BDB_LIBFILES) ../libical/libical_cxx.la
 libicalss_cxx_la_SOURCES =	\
-	$(libicalss_evolution_la_SOURCES)	\
+	$(libicalss_la_SOURCES)	\
 	icalspanlist_cxx.h	\
 	icalspanlist_cxx.cpp
 
--- origsrc/libical-0.31/src/libicalss/icalbdbset.c	2008-03-12 20:50:05.000000000 -0500
+++ src/libical-0.31/src/libicalss/icalbdbset.c	2008-05-04 02:47:26.078125000 -0500
@@ -1302,7 +1302,7 @@
 
             /* if it is excluded, do next one */
             if (icalproperty_recurrence_is_excluded(comp, &start, &next)) {
-	        icalrecur_iterator_decrement_count(itr.ritr);
+	        icalrecur_iterator_next(itr.ritr);
                 continue;
             }
 
@@ -1409,7 +1409,7 @@
 
     /* if it is excluded, return NULL to the caller */
     if (icalproperty_recurrence_is_excluded(comp, &start, &next)) {
-        icalrecur_iterator_decrement_count(itr->ritr);
+        icalrecur_iterator_next(itr->ritr);
        return NULL; 
     }
 
@@ -1515,7 +1515,7 @@
 
             /* if it is excluded, do next one */
             if (icalproperty_recurrence_is_excluded(comp, &start, &next)) {
-                icalrecur_iterator_decrement_count(i->ritr);
+                icalrecur_iterator_next(i->ritr);
                 continue;
             }
 
--- origsrc/libical-0.31/src/libicalvcal/Makefile.am	2008-03-12 20:50:06.000000000 -0500
+++ src/libical-0.31/src/libicalvcal/Makefile.am	2008-05-04 02:21:16.453125000 -0500
@@ -8,8 +8,8 @@
 	-I$(top_builddir)/src/libical	\
 	-I$(top_srcdir)/src/libicalss
 
-libicalvcal_la_LDFLAGS = -version-info 30:0:30
-libicalvcal_la_LIBADD = ../libical/.libs/libical.la
+libicalvcal_la_LDFLAGS = -version-info 30:0:30 -no-undefined
+libicalvcal_la_LIBADD = ../libical/libical.la
 
 if DEV
 libicalvcal_la_DEVSOURCES = vcc.y
--- origsrc/libical-0.31/src/test/regression-storage.c	2008-03-12 20:50:02.000000000 -0500
+++ src/libical-0.31/src/test/regression-storage.c	2008-05-04 02:54:05.281250000 -0500
@@ -68,6 +68,7 @@
 
 #ifdef WITH_BDB
 #include <db.h>
+#include <libicalss/icalbdbset.h>
 
 int get_title(DB *dbp, const DBT *pkey, const DBT *pdata, DBT *skey);
 char * parse_vcalendar(const DBT *dbt) ;
